lib/explorer/polars_backend/shared.ex

```diff
@@ -186,9 +186,30 @@ defmodule Explorer.PolarsBackend.Shared do
       {:duration, :nanosecond} ->
         Native.s_from_binary_i64(name, binary) |> Native.s_cast(dtype) |> ok()
 
+      {:s, 8} ->
+        Native.s_from_binary_i8(name, binary)
+
+      {:s, 16} ->
+        Native.s_from_binary_i16(name, binary)
+
+      {:s, 32} ->
+        Native.s_from_binary_i32(name, binary)
+
       {:s, 64} ->
         Native.s_from_binary_i64(name, binary)
```

Comment:
I wonder if it is worth having a single function where we pass the integer or the size as argument? Not sure what is easier on the Rust side.